apps/users/models.py

```diff
@@ -145,6 +150,13 @@ def create_superuser(self, username, email, password):
         GroupUser.objects.create(user=user, group=admins)
         return user
 
+    def _generate_username(self, seed):
+        if self.model.objects.filter(username=seed).exists():
+            return '{seed}-{num}'.format(
+                seed=seed, num=random.randint(1000, 9999))
```

Comment:
It's a pretty sad story for a user who tries the process and it fails randomly though. And you are going to love tracking down that bug 10 months later.

Also log the crap out of this.